src/services/codefixes/inferFromUsage.ts

```diff
@@ -731,12 +760,16 @@ namespace ts.codefix {
 
         function inferFromContext(usageContext: UsageContext, checker: TypeChecker) {
             const types = [];
+
             if (usageContext.isNumber) {
                 types.push(checker.getNumberType());
             }
-            if (usageContext.isString) {
+            else if (usageContext.isString) {
                 types.push(checker.getStringType());
             }
+            else if (usageContext.isNumberOrString) {

```

Comment:
Isn't the `stringNumber` priority unnecessary since we're using `else if` here?

Reply:
I don't understand what you mean here. Is it irrelevant with the switch from `else if` to `if` anyway?